src/main/java/com/github/jrubygradle/jem/JemsClassLoader.java

```diff
@@ -0,0 +1,50 @@
+package com.github.jrubygradle.jem;
+
+import com.github.jrubygradle.jem.internal.GemInstaller;
+import com.github.jrubygradle.jem.GemInstaller.DuplicateBehavior;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.util.Collections;
+
+/**
+ * Created by cmeier on 8/15/15.
+ */
+public class JemsClassLoader extends URLClassLoader {
+
+    private final File jemsDir;
+
+    public static JemsClassLoader create() throws IOException {
+        return new JemsClassLoader(Files.createTempDirectory("jems").toFile());
+    }
+
+    public JemsClassLoader(File jemsDir) throws IOException {
+        super(new URL[] { jemsDir.toURI().toURL() });
+        this.jemsDir = jemsDir;
+    }
+
+    public Gem addJem(String jem) throws IOException {
+        return addJem(new File(jem));
+    }
+    public Gem addJem(File jem) throws IOException {
```

Comment:
I understand why this still has to be here and we have to stuff things into a temporary directory. Is there any way to pass a VFS type object up to the rest o JRuby such that we can just look inside our own memory space for this gem content?